Migration to AsciiDoc (#549)
OpenSimulationInterface/open-simulation-interface: doc/usecases/transforming_coordinate_systems.adoc

> = Transforming coordinate systems

Reviewer: ```suggestion
= Coordinate Transformations
```

Author: I would stick to a more active tone, similar to "Modelling a traffic participant".

> Transforming coordinates for a specific vehicle or sensor is a common task when running simulations.

Reviewer: ```suggestion
Coordinate transformation for a specific vehicle or sensor is a common task when running simulations.
```

Author: see above, I prefer a more active tone.

OSI uses DIN ISO 8855:2013-11 for coordinate systems and transformations between coordinate systems.

> //TODO: Should we add one or more sentences about the mathematical operations involved?

Reviewer: @max-rosin, are they mentioned elsewhere, if yes we should add a link to them?
If no then, yes we should populate this section with more data.

Author: No, this was more of an idea I had while reading the topic. I was especially thinking about the order of rotations.

Reviewer 2: CCB: Should be tackled after migration is done.

**Corresponding messages**

``GroundTruth::moving_object::base::position``::
Defines the position of the vehicle’s reference point in world coordinates.

> In Open Simulation Interface, the position of an object is defined by the coordinates of the center of the object's bounding box.

Reviewer: "Object coordinate system:
Coordinate system for local object coordinates. 
The origin of the corresponding coordinate systems is not necessarily identical to the center of the object's bounding box.
For example with vehicles, the origin is the middle of the rear axle projected onto the ground."
@pmai we should discuss this in our next CCB meeting.

Author: I am not sure what change you are suggesting. Sounds like a quote from the coordinate-system topic.

Reviewer 3: @pasched to be done

Reviewer 3: @pasched to be done.

Reviewer 3: OSI CCB: 
- review in CCB: 
```suggestion
In Open Simulation Interface, the position of an object is defined by the coordinates of the center of the object's 3D bounding box.
```

Reviewer 3: @pasched Todo

``GroundTruth::moving_object::base::orientation``::
Defines the orientation of the vehicle’s reference point in world coordinates.

``GroundTruth::moving_object::vehicle_attributes::bbcenter_to_rear``::
The vector pointing from the vehicle’s reference point to the middle of the rear axle under neutral load conditions in vehicle coordinates.

> 

Reviewer: @ThomasNaderBMW, @pmai, @clemenshabedank 
What about other objects, like stationary objects, traffic signs and lights?
To be discussed in the next CCB meeting.

Reviewer 3: @pasched to be done. To be discussed in CCB today.

Reviewer 3: OSI CCB:
- Review in OSI CCB: Should be fine like this.

``SensorData::mounting_position``::

> Defines the sensor’s position and orientation and thereby the origin of the sensor coordinate system.

Reviewer: ```suggestion
Defines the sensor’s position and orientation and thereby the origin of the sensor's coordinate system.
```

Author: This was intentional. I was referring to the _sensor coordinate system_, see https://github.com/OpenSimulationInterface/open-simulation-interface/blob/migration/doc/architecture/reference_points_coordinate_systems.adoc

The mounting position is given in the vehicle coordinate system.

**Example**

The following image shows the relationship between the coordinate systems.
The reference point of the vehicle is given by a vector in the world coordinate system.
A vector pointing from the reference position of the vehicle to the center of the rear axle then yields the origin of the vehicle coordinate system.
The mounting positions of the sensors and therefore the origins of the corresponding sensor coordinate systems are then given with respect to the vehicle coordinate system.

> image::{images_open_simulation_interface}/osi_example_coordinate_systems.png[]

Reviewer 3: Please review @kmeids
